src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -900,6 +867,17 @@ private sealed class NullStream : Stream
                 get { return 0; }
                 set {}
             }
+            
+            public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+            {
+                // Validate arguments here for compat, since previously this method
+                // was inherited from Stream (which did check its arguments).
+                ValidateCopyToArguments(destination, bufferSize);
+                
+                if (cancellationToken.IsCancellationRequested)
+                    return Task.FromCancellation(cancellationToken);
+                return Task.CompletedTask;
```

Comment:
This could be simplified:

```csharp
return cancellationToken.IsCancellationRequested ? Task.FromCancellation(cancellationToken) : Task.CompletedTask;
```